feat(BoaBua): add source BaoBua

marioplus opened the pull request:
Closes: #1104

Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [x] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [ ] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio
- [x] Have removed `web_hi_res_512.png` when adding a new extension

marioplus committed:
feat(buondua): add source BaoBua

Refs: #1104

vetleledaal commented on src/all/baobua/src/eu/kanade/tachiyomi/extension/all/baobua/BaoBua.kt:
```diff
@@ -0,0 +1,111 @@
+    override val baseUrl = "https://www.baobua.net"
+    override val lang = "all"
+    override val name = "BaoBua"
+    override val supportsLatest = true
```
It is assumed sites have a popular listing always, in this case it appears to only have a latest listing. By convention the latest listing has been put in the popular tab if it is the only one available.
```suggestion
    override val supportsLatest = false
```

vetleledaal commented on src/all/baobua/src/eu/kanade/tachiyomi/extension/all/baobua/BaoBua.kt:
```diff
@@ -0,0 +1,111 @@
+    override fun chapterFromElement(element: Element) = SChapter.create().apply {
+        chapter_number = 0F
+        setUrlWithoutDomain(element.selectFirst("div.breadcrumb-trail li.trail-end > a")!!.absUrl("href"))
+        date_upload = POST_DATE_FORMAT.tryParse(element.selectFirst("span.item-metadata.posts-date")?.text()?.trim())
```
`.text()` is by definition already trimmed.
```suggestion
        date_upload = POST_DATE_FORMAT.tryParse(element.selectFirst("span.item-metadata.posts-date")?.text())
```

vetleledaal commented on src/all/baobua/src/eu/kanade/tachiyomi/extension/all/baobua/BaoBua.kt:
```diff
@@ -0,0 +1,111 @@
+    override fun mangaDetailsParse(document: Document): SManga {
+        val trailItemsEl = document.selectFirst("div.breadcrumb-trail > ul.trail-items")!!
+        return SManga.create().apply {
+            title = trailItemsEl.selectFirst("li.trail-end")!!.text().trim()
```
Already trimmed.
```suggestion
            title = trailItemsEl.selectFirst("li.trail-end")!!.text()
```

vetleledaal commented on src/all/baobua/src/eu/kanade/tachiyomi/extension/all/baobua/BaoBua.kt:
```diff
@@ -0,0 +1,111 @@
+        return filter.selectedCategory?.let {
+            GET(it.url, headers)
+        } ?: run {
+            "$baseUrl/".toHttpUrl().newBuilder()
```
Is the trailing slash here necessary? Try:
```suggestion
            baseUrl.toHttpUrl().newBuilder()
```

marioplus committed:
fix(buondua/search): resolve pagination param forwarding
chore(buondua/search): clean up URL format and flag

* Fix:
  - Add missing pagination parameter propagation
* Maintenance:
  - Remove redundant trailing "/" in pagination URLs
  - Set `supportsLatest` to false (default behavior)

Closes: #1104

marioplus committed:
chore(buondua): remove redundant trim() calls

Closes: #1104

marioplus committed:
Merge branch 'main' into baobua

AwkwardPeak7 commented on settings.gradle.kts:
revert

AwkwardPeak7 commented on src/all/baobua/src/eu/kanade/tachiyomi/extension/all/baobua/BaoBua.kt:
```diff
@@ -0,0 +1,111 @@
+    // endregion
+
+    // region latest
+    override fun latestUpdatesRequest(page: Int) = throw Exception("Stub!")
```
throw empty `UnsupportedOperationException`

AwkwardPeak7 commented on src/all/baobua/src/eu/kanade/tachiyomi/extension/all/baobua/BaoBua.kt:
```diff
@@ -0,0 +1,111 @@
+        chapter_number = 0F
+        setUrlWithoutDomain(element.selectFirst("div.breadcrumb-trail li.trail-end > a")!!.absUrl("href"))
+        date_upload = POST_DATE_FORMAT.tryParse(element.selectFirst("span.item-metadata.posts-date")?.text())
+        name = DISPLAY_DATE_FORMAT.format(Date(date_upload))
```
imo just set "Gallery" as name

AwkwardPeak7 commented on src/all/baobua/src/eu/kanade/tachiyomi/extension/all/baobua/SourceCategorySelector.kt:
```diff
@@ -0,0 +1,47 @@
+                SourceCategory("hot", "c3VRb3RJZ2wrU2tTYmpGSUVqMnFndz09"),
+                SourceCategory("big breast", "dkQ3b0RiK0xpZDRlMVNSY3lUNkJXQT09"),
+            ).onEach {
+                it.url = "$baseUrl/".toHttpUrl().newBuilder()
```
build url in `searchMangaRequest` and dont save baseurl in filter

marioplus committed:
refactor(BaoBua): standardize chapter names and URL handling

- Replace date-based chapter names with static "Gallery" value
- Remove baseUrl from category URLs (construct dynamically when used)

Closes: #1104

marioplus committed:
chore(BaoBua): remove unused

Closes: #1104

marioplus committed:
chore(BaoBua): revert settings.gradle.kts

Closes: #1104

marioplus committed:
chore(BaoBua): remove unused import

Closes: #1104

marioplus committed:
chore(BaoBua): remove needless blank line

Closes: #1104

marioplus committed:
fix(BaoBua): add unselected Category

Closes: #1104

AwkwardPeak7 commented on src/all/baobua/src/eu/kanade/tachiyomi/extension/all/baobua/BaoBua.kt:
```diff
@@ -0,0 +1,107 @@
+        val trailItemsEl = document.selectFirst("div.breadcrumb-trail > ul.trail-items")!!
+        return SManga.create().apply {
+            title = trailItemsEl.selectFirst("li.trail-end")!!.text()
+            genre = trailItemsEl.select("li:not(.trail-end):not(.trail-begin)").joinToString { it.text() }
```
also set `update_strategy = UpdateStrategy.ONLY_FETCH_ONCE`

AwkwardPeak7 commented on src/all/baobua/build.gradle:
```diff
@@ -0,0 +1,12 @@
+
+dependencies {
+    implementation(project(":lib:randomua"))
+}
```
remove since not used in the source class

marioplus committed:
refactor(BaoBua): optimize manga details parsing

- Set update_strategy = UpdateStrategy.ONLY_FETCH_ONCE
- Remove unused randomua dependency

Closes: #1104
